code/game/machinery/autolathe.dm

```diff
@@ -219,3 +219,32 @@
 			qdel(I)
 			return 1
 	return 0
+
+/obj/machinery/r_n_d/fabricator/mechanic_fab/autolathe/MouseDropTo(atom/over_object, mob/user)
+	if(istype(over_object, /obj/item/weapon/storage/bag/gadgets/part_replacer))
+		if(user.incapacitated() || !user.Adjacent(src))
+			return
+		var/obj/item/weapon/storage/bag/gadgets/part_replacer/partReplacer = over_object
+		if(partReplacer.contents.len <= 0)
+			return 0
+		for(var/obj/item/weapon/stock_parts/stockParts in partReplacer.contents)
+			if(!stockParts.materials)
+				continue
+			else if(stockParts.materials.getVolume() + src.materials.getVolume() > max_material_storage)
+				to_chat(user, "\The [src]'s material bin is too full to continue recycling.")
+				return 0
+			else if(allowed_materials)
+				var/allowed_materials_volume = 0
+				for(var/mat_id in allowed_materials)
+					allowed_materials_volume += stockParts.materials.storage[mat_id]
+				if (allowed_materials_volume != stockParts.materials.getVolume())
+					continue
+			else if(!stockParts.recyclable())
+				continue
+			materials.removeFrom(stockParts.materials)
+			qdel(stockParts)
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
```

Comment:
```suggestion
		playsound(src, 'sound/machines/click.ogg', 50, 1)
```

Reply:
does that really work? I ctrl+f'd and saw a few instances of the get_turf(src) instead of src.
I fear that doing src will play it like a local sound or something?

Reply:
the first thing `playsound` does is `get_turf()` on the first parameter.
[I once removed all instances of that](https://github.com/vgstation-coders/vgstation13/pull/17640) but people keep doing it